### PR TITLE
Sync demo

### PR DESCRIPTION
**Pull request by secondsun**
How to test

1. Clone and build locally [Android Sync Library](https://github.com/aerogear/aerogear-android-sync/tree/master/aerogear-android-sync)
1. Run the [Sync Server (JSON Patch)](https://github.com/aerogear/aerogear-sync-server)
1. Configure the Application
   1. Set serverHost in AndroidManifest.xml to an appropriate host. The default is for the emulator host's IP (so if you run this in the Android emulator you will get the desired behavior)
1. Run the app (./gradlew installDebug)
1. Edit the doc then Save
1. Close the app (or even uninstall)
1. run the app again
1. you should see your changes

Bonus testing!

Multiple devices. You should see changes on save propagate.

### CODE REVIEW

File: SyncDemo/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+dependencies {
+    compile 'com.android.support:appcompat-v7:21.0.3'
+    compile 'com.android.support:multidex:1.0.0'
+    compile 'com.melnykov:floatingactionbutton:1.1.0'    
+    compile 'org.jboss.aerogear:aerogear-android-core:2.0.0'
+    compile 'org.jboss.aerogear:aerogear-android-pipe:2.0.0'
+    compile 'org.jboss.aerogear:aerogear-android-store:2.0.0'
+    compile 'org.jboss.aerogear:aerogear-android-authz:2.0.0'
+    compile (group:'org.jboss.aerogear', name:'sync-client-android', version:'0.0.1-SNAPSHOT', changing:'true' , ext:'aar')
```
Comment (matzew) [inline SyncDemo/app/build.gradle]: where is that library located?

Comment (danielpassos) [inline SyncDemo/app/build.gradle]: @matzew https://github.com/aerogear/aerogear-android-sync/tree/master/aerogear-android-sync

Comment (matzew) [inline SyncDemo/app/build.gradle]: but isn't that a different artifact ID ?

https://github.com/aerogear/aerogear-android-sync/blob/master/aerogear-android-sync/pom.xml#L23

On Thu, Mar 5, 2015 at 12:35 PM, Daniel Passos <notifications@github.com>
wrote:

> In SyncDemo/app/build.gradle
> <https://github.com/aerogear/aerogear-android-cookbook/pull/49#discussion_r25857149>
> :
>
> > +        exclude 'META-INF/ASL-2.0.txt'
> > +        exclude 'META-INF/LGPL-3.0.txt'
> > +        exclude 'META-INF/io.netty.versions.properties'
> > +    }
> > +
> > +}
> > +
> > +dependencies {
> > +    compile 'com.android.support:appcompat-v7:21.0.3'
> > +    compile 'com.android.support:multidex:1.0.0'
> > +    compile 'com.melnykov:floatingactionbutton:1.1.0'
> > +    compile 'org.jboss.aerogear:aerogear-android-core:2.0.0'
> > +    compile 'org.jboss.aerogear:aerogear-android-pipe:2.0.0'
> > +    compile 'org.jboss.aerogear:aerogear-android-store:2.0.0'
> > +    compile 'org.jboss.aerogear:aerogear-android-authz:2.0.0'
> > +    compile (group:'org.jboss.aerogear', name:'sync-client-android', version:'0.0.1-SNAPSHOT', changing:'true' , ext:'aar')
>
> @matzew <https://github.com/matzew>
> https://github.com/aerogear/aerogear-android-sync/tree/master/aerogear-android-sync
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-android-cookbook/pull/49/files#r25857149>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

Comment (secondsun) [inline SyncDemo/app/build.gradle]: Good catch

**Comment by cvasilak** [comment]
tested example with both iOS and Javascript demos and worked great, no issues.

In fact have prepared [a small screencast](https://vimeo.com/121332828) demonstrate all three platforms in-action going against latest sync-server 'master' snapshot.

:+1:

**Comment by danbev** [comment]
@cvasilak Nice screen cast!

**Comment by danielpassos** [comment]
:+1:

**Comment by abstractj** [comment]
@secondsun landed

**Pull request closed without merging**